include/deal.II/grid/grid_refinement.h

```diff
@@ -48,6 +48,41 @@ template <class T> class Vector;
 namespace GridRefinement
 {
   /**
+   * This function adjust cell number fraction of refinement and coarsening
+   * according to current cell number and maximal cell number.
+   *
+   * @arg @p current_n_cells is current cell number.
+   *
+   * @arg @p max_n_cells is specified a maximal number of cells. If current
+   * cell number is already exceeded maximal cell number, @p max_n_cells will
+   * be set to zero and @p bottom_fraction will be adjusted to reduce cell
+   * number to @ max_n_cells. If cell number is going to be exceeded only
+   * upon refinement, then refinement and coarsening fractions are going
+   * to be adjusted with a same ratio in an attempt to reach the maximum
+   * number of cells. Be aware though that through proliferation of refinement
+   * due to Triangulation::MeshSmoothing, this number is only an indicator.
+   * The default value of this argument is to impose no limit on the number
+   * of cells.
+   *
+   * @arg @p top_fraction is required refinement fraction. It's value may be
+   * changed after this function call.
+   *
+   * @arg @p bottom_fraction is required coarsening fraction. It's value may be
+   * changed after this function call.
+   *
+   * @note Usually you do not need to call this function explicitly. Pass
+   * @p max_n_cells to function refine_and_coarsen_fixed_number or function
+   * refine_and_coarsen_fixed_fraction and they will call this function if
+   * necessary.
+   */
+  template <int dim>
+  void
+  adjust_refine_and_coarsen_number_fraction (const unsigned int  current_n_cells,
+                                             const unsigned int  max_n_cells,
+                                             double             &top_fraction,
+                                             double             &bottom_fraction);
```

Comment:
inout arguments are often confusing to the reader, in particular if there are more than one. It is true that you need the temporary at the calling location, though you can avoid this using `boost::tie`: http://www.boost.org/doc/libs/1_58_0/libs/tuple/doc/tuple_users_guide.html#tiers

I would prefer the change to return values by value, rather than through a reference argument.